modelmesh-monitoring/base/mesh-monitoring-stack.yaml

```diff
@@ -0,0 +1,127 @@
+---
+apiVersion: operators.coreos.com/v1alpha2
+kind: OperatorGroup
+metadata:
+  name: rhods-monitoring-og
+  namespace: redhat-ods-monitoring
+spec:
+  targetNamespaces:
+    - redhat-ods-monitoring
+
+---
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: rhods-prometheus
+  namespace: redhat-ods-monitoring
+spec:
+  channel: beta
+  name: rhods-prometheus-operator
+  source: addon-managed-odh-catalog
+  sourceNamespace: openshift-marketplace
+
+---
+kind: ServiceAccount
+apiVersion: v1
+metadata:
+  annotations:
+    serviceaccounts.openshift.io/oauth-redirectreference.prometheus: '{"kind":"OAuthRedirectReference","apiVersion":"v1","reference":{"kind":"Route","name":"rhods-model-monitoring"}}'
+  name: rhods-model-monitoring-sa
+  namespace: redhat-ods-monitoring
+
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: rhods-model-monitoring-crb
+subjects:
+  - kind: ServiceAccount
+    name: rhods-model-monitoring-sa
+    namespace: redhat-ods-monitoring
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: prometheus-scraper
+
+---
+kind: Service
+apiVersion: v1
+metadata:
+  name: rhods-model-monitoring
+  annotations:
+    service.alpha.openshift.io/serving-cert-secret-name: serving-prometheus-proxy-tls
+  labels:
+    prometheus: rhods-model-monitoring
+spec:
+  ports:
+    - name: https
+      protocol: TCP
+      port: 443
+      targetPort: 8443
+  selector:
+    prometheus: rhods-model-monitoring
+
+---
+apiVersion: monitoring.coreos.com/v1
+kind: Prometheus
+metadata:
+  labels:
+    prometheus: rhods-model-monitoring
+  name: rhods-model-monitoring
+spec:
+  serviceAccountName: rhods-model-monitoring-sa
+  listenLocal: true
+  containers:
+    - args:
+        - '-https-address=:8443'
+        - '-provider=openshift'
+        - '-openshift-service-account=rhods-model-monitoring-sa'
+        - '-upstream=http://localhost:9090'
+        - '-tls-cert=/etc/tls/private/tls.crt'
+        - '-tls-key=/etc/tls/private/tls.key'
+        - '-cookie-secret=SECRET' # TO DO, FIX THIS. USE DEPLOYER TO GENERATE A SECRET THAT IS MOUNTED HERE
```

Comment:
This secret can be randomized using the secret generator controller, see https://github.com/red-hat-data-services/odh-manifests/blob/master/odh-dashboard/overlays/authentication/secret.yaml#L1-L10